### PR TITLE
Fixed a couple breaking issues

### PR DESCRIPTION
**Pull request by benmarch**
This is a fantastic library, but I had some issues after installing it:

* The response interceptor returns just a Resource, which does not contain all of the properties from the response. This is a problem because another plugin that I am using looks for the `config` property, and throws an error. I haven't fully tested my change, but it looks like adding the properties to the resource as non-enumerable fixes the problem with no side effects. I wanted to keep this change backwards-compatible, otherwise I would have returned the response and made `response.data` the Resource.
* This might have been caused by our Webpack config, but we were getting build errors when importing. I think it is because I am excluding anything under `node_modules` from Babel. I just changed package.json to point to the dev bundle.

Thank you for creating and maintaining this! Saved me so much time!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     }
   ],
   "license": "WTFPL",
-  "main": "src/index.js",
+  "main": "dist/angular-hal.js",
```
Comment (maennchen) [inline package.json]: The main property has to point to the index.js and not a generated dist file. If this doesn't work with your webpack config, I recommend changing the config.

File: package.json [hunk 2]
```diff
@@ -46,9 +46,9 @@
     "webpack": "^1.13.0"
   },
   "scripts": {
-    "build:dev": "./node_modules/.bin/browserify src/index.js -o dist/angular-hal.map.js  -t [ babelify --presets [ es2015 ] ] --s angularHal -d",
```
Comment (maennchen) [inline package.json]: Why did you change the file paths?

**Review by maennchen — CHANGES_REQUESTED** [review]
I don't like having those properties inside of the resource, even if they are non-enumerable. I'd rather create a breaking change that for example adds the resource onto the respone.

**Comment by benmarch** [comment]
Hey @maennchen, thank you for reviewing so quickly. I agree, I don't like the approach, I would be happy to change it so that the response is in tact but the data is the resource. I changed the file path because the dev file is more than just a source map file (no functional reason though).

I'll revert these changes and just submit the change to the interceptor. As you mentioned, I can get around the Webpack issue on my end, so that change is not really necessary (nor the filename change).

**Comment by agouriou** [comment]
Hello,

Just to understand: @benmarch you did this change because another plugin setup another http interceptor  that requires `config` ?

**Comment by benmarch** [comment]
Hey @agouriou, yes, that is exactly right. I'll have the changes in later today for your review.

**Comment by agouriou** [comment]
What I did in a similar case was to register the interceptors in an order so that angular-hal's intercepter is the last one to operate. 

By doing so, all interceptors worked as expected since the initial `Response` object was only transformed to a `Resource` in the last step of interception.

Maybe this could suit your needs...

**Comment by benmarch** [comment]
That was my first approach but since I don't have control over when either interceptor is registered, I couldn't figure out how to ensure the order. Did you have to manually order `$httpProvider.interceptors` in your config block?

I would rather not use this solution since it is still a bit of a hack. I think it is reasonable for a library to assume that a response object will be passed into a response interceptor, so I think it is best for angular-hal to uphold that assumption. I am setting up my changes in a way where if you use `$http` to request a resource then you will get the response object, but if you use `halClient` then you will get the resource as you do now. It should require very minimal changes to consumer code.

**Comment by maennchen** [comment]
I think it would be a reasonable decision to move the resource into the response data and return the whole response object. This should suit both needs.

The reason I didn't implement it this way, is that I want the same interface for links and embeds. With embeds there would be no normal response since it requires no HTTP request.

If you can come up with a good solution to that problem, I'd be happy to accept a PR.

**Comment by benmarch** [comment]
@agouriou @maennchen ready for your review. You can see from the changes to the test cases that consuming code will only have to make changes when using `$http` to request a resource. If developers followed your examples then it is likely that they only need to make a single change when requesting the API root resource. The `halResourceClient` and `halClient` APIs are unchanged, so there is still consistency between getting links and embedded resources.

**Comment by maennchen** [comment]
I'm currently on vacation. I'll have a thoroughly look at it when I'm back home.

It looks good on first glance, thank you verry much.

**Review by maennchen — APPROVED** [review]

**Comment by benmarch** [comment]
Thank you, @maennchen! Please let me know when I can update my app with the new version.

**Comment by maennchen** [comment]
Could you create a migration guide and include it in the readme?

**Comment by benmarch** [comment]
Sure, will this be a major release (3.0.0)?

**Comment by maennchen** [comment]
Yes

**Comment by benmarch** [comment]
Created #70 with updates to README.

**Pull request merged**